### PR TITLE
sched/assert: Fix printing argv when address environments are in use

### PR DESCRIPTION
**Pull request by pussuw**
## Summary
Instantiate the correct address environment when reading the process's argument vector. Otherwise doing this will crash the system every time, causing a recursive assert loop.

Also try to do a bit of sanity checking before attempting to read the process's memory, it might be in a bad state in which case this will fail anyway.
## Impact
Fixes a recursive assert loop, when address environments are in use
## Testing
icicle:knsh

### CODE REVIEW

File: sched/misc/assert.c [hunk 1]
```diff
@@ -209,11 +209,78 @@ static void showstacks(void)
...
+ *   Safely read the contents of a task's argument vector, into a a safe
+ *   buffer.
+ *
+ ****************************************************************************/
+
+static void get_argv_str(FAR struct tcb_s *tcb, FAR char *args, size_t size)
+{
+  /* Perform sanity checks */
+
+  if (!tcb || !tcb->group || !tcb->group->tg_info)
```
Comment (pkarashchenko) [inline sched/misc/assert.c:225]: Optional
```suggestion
  if (tcb == NULL || tcb->group == NULL || tcb->group->tg_info == NULL)
```

File: sched/misc/assert.c [hunk 2]
```diff
@@ -209,11 +209,78 @@ static void showstacks(void)
...
+    {
+      /* Something is very wrong -> get out */
+
+      *args = '\0';
+      return;
+    }
+
+#ifdef CONFIG_ARCH_ADDRENV
+  save_addrenv_t oldenv;
+  bool saved = false;
```
Comment (pkarashchenko) [inline sched/misc/assert.c:235]: not compatible with C89 and should be moved to the top of the function

File: sched/misc/assert.c [hunk 3]
```diff
@@ -209,11 +209,78 @@ static void showstacks(void)
...
+
+      up_addrenv_select(&tcb->group->tg_addrenv, &oldenv);
+      saved = true;
+    }
+#endif
+
+#ifndef CONFIG_DISABLE_PTHREAD
+  if ((tcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD)
+    {
+      struct pthread_tcb_s *ptcb = (struct pthread_tcb_s *)tcb;
```
Comment (pkarashchenko) [inline sched/misc/assert.c:255]:
```suggestion
      FAR struct pthread_tcb_s *ptcb = (FAR struct pthread_tcb_s *)tcb;
```

File: sched/misc/assert.c [hunk 4]
```diff
@@ -209,11 +209,78 @@ static void showstacks(void)
...
+
+      snprintf(args, size, " %p %p", ptcb->cmn.entry.main, ptcb->arg);
+    }
+  else
+#endif
+    {
+      char **argv = tcb->group->tg_info->argv + 1;
```
Comment (pkarashchenko) [inline sched/misc/assert.c:262]:
```suggestion
      FAR char **argv = tcb->group->tg_info->argv + 1;
```

File: sched/misc/assert.c (hunk 3 above)
Comment (pussuw) [reply sched/misc/assert.c:255]: Thanks, I fixed a couple of missing FAR qualifiers but missed these (this code is copy&paste).

**Pull request merged**